sdk/tracing/tracing-opentelemetry/src/main/java/com/azure/tracing/opentelemetry/OpenTelemetryTracer.java

```diff
@@ -23,34 +31,57 @@
     private static final Tracer TRACER = Tracing.getTracer();
     private static final String OPENTELEMETRY_SPAN_KEY = com.azure.core.implementation.tracing.Tracer.OPENTELEMETRY_SPAN_KEY;
     private static final String OPENTELEMETRY_SPAN_NAME_KEY = com.azure.core.implementation.tracing.Tracer.OPENTELEMETRY_SPAN_NAME_KEY;
+    private static final String DIAGNOSTIC_ID = com.azure.core.implementation.tracing.Tracer.OPENTELEMETRY_DIAGNOSTIC_ID_KEY;
+    private static final String ENTITY_PATH = com.azure.core.implementation.tracing.Tracer.OPENTELEMETRY_AMQP_ENTITY_PATH;
+    private static final String HOSTNAME = com.azure.core.implementation.tracing.Tracer.OPENTELEMETRY_AMQP_HOST_NAME;
+    private static final String SPAN_CONTEXT = com.azure.core.implementation.tracing.Tracer.OPENTELEMETRY_AMQP_EVENT_SPAN_CONTEXT;
+
+    // standard attributes with AMQP call information
+    private static final String COMPONENT = "component";
+    private static final String MESSAGE_BUS_DESTINATION = "message_bus.destination";
+    private static final String PEER_ENDPOINT = "peer.address";
 
     private final ClientLogger logger = new ClientLogger(OpenTelemetryTracer.class);
 
     @Override
-    public Context start(String methodName, Context context) {
-        Span parentSpan = (Span) context.getData(OPENTELEMETRY_SPAN_KEY).orElse(TRACER.getCurrentSpan());
-        String spanName = (String) context.getData(OPENTELEMETRY_SPAN_NAME_KEY).orElse(methodName);
-
-        SpanBuilder spanBuilder = TRACER.spanBuilderWithExplicitParent(spanName, parentSpan);
-        Span span = spanBuilder.startSpan();
-
+    public Context start(String spanName, Context context) {
+        Span span = startSpanWithExplicitParent(spanName, context);
+        if (context.getData(ENTITY_PATH).isPresent() && span.getOptions().contains(Span.Options.RECORD_EVENTS)) {
+            // If span is sampled in, add additional TRACING attributes
+            addSpanRequestAttributes(span, context, spanName);
+        } else {
+            // Add diagnostic Id to Context
+            context = setContextData(span);
```

Comment:
I think I'm missing something

we start span from explicit parent - fine
if there is entity path, we add it to span - fine
why we setContextData if there is no entity path OR span is sampled out? - this doesn't look right

Reply:
I see, you distinguish 'message' from 'send' span this way. Is there a more obvious and stable way?

Reply:
I think we can directly use the span name to distinguish that. It would be well-formed by the time it reaches here. `Azure.eventhubs.send` and `Azure.eventhubs.message`

Reply:
can we do this without relying on the span name? E.g. explicitly call setContextData when needed? 
It seems like if span name changes or we need to add other kids of spans, we'd have special-case all names here. 
This is also a bit hard to read and follow.